wcfsetup/install/files/lib/system/option/LineBreakSeparatedTextOptionType.class.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace wcf\system\option;
+
+use wcf\data\option\Option;
+use wcf\system\WCF;
+use wcf\util\ArrayUtil;
+use wcf\util\StringUtil;
+
+/**
+ * Option type implementation for separate items that are stored as line break-separated text.
+ *
+ * @author  Matthias Schmidt
+ * @copyright   2001-2021 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Option
+ * @since   5.4
+ */
+class LineBreakSeparatedTextOptionType extends TextareaOptionType
+{
+    /**
+     * @inheritDoc
+     */
+    public function getFormElement(Option $option, $value)
+    {
+        $values = ArrayUtil::trim(\explode("\n", StringUtil::unifyNewlines($value)));
+        \uasort($values, 'strnatcmp');
```

Comment:
Did you verify how the sorting behaves with non-ASCII values?

Reply:
It is now consistent between JavaScript and PHP. We generally use the the `str*cmp` functions when sorting values.